src/cli/workflows.ts

```diff
@@ -173,16 +174,20 @@ export const wf = () => {
     .addOption(options.name)
     .addOption(options.tag)
     .addOption(options.excludeId)
+    .addOption(options.activate)
     .action(createAction(async (opts, wf, cmd) => {
+      const wfFilter = getWfFilter(opts, config);
       const args: Parameters<typeof wf.publish> = [
         opts.dir || config.workflows.dir,
-        getWfFilter(opts, config)
-      ]
-      logOp(cmd, args)
+        wfFilter,
+        opts.activate
+      ];
+      logOp(cmd, args);
       if (opts.dry === false) {
-        await wf.publish(...args)
+        await wf.deactivate(wfFilter) // Deactivate to avoid conflict
```

Comment:
Can you please clarify why we need to deactivate workflows before publishing? The comment, BTW, doesn't do it.

Reply:
We deactivate the workflow before publishing to avoid conflicts that can arise when activating a workflow that might already have active triggers or schedules, which n8n doesn’t handle well. More details about this issue can be found here: [Workflow could not be activated](https://community.n8n.io/t/workflow-could-not-be-activated/19621).